### DISCUSSION TITLE
Disable fillable PDFs / forms

### DISCUSSION THREAD

**Discussion by wjmackinnon** [stephanrauh/ngx-extended-pdf-viewer#1663] [Q&A]
Hello, I am looking to disable fillable PDFs with this module. I can only find out how to change the theme etc. Any help would be greatly appreciated!

**Comment by stephanrauh** [comment]
The forms are simple HTML forms, so you can disable them yourself.

- Catch the event `(annotationLayerRendered)` (requires the most current version of ngx-extended-pdf-viewer 16.0.0-alpha.5)
- The event is called for each page. The page number is part of the event. The `source` attribute contains a reference to the div.
- Disable every input element of this div (plus radiobuttons, textareas, and `<select>` elements):

```html
<ngx-extended-pdf-viewer (annotationLayerRendered)="onEvent($event)">
```

```ts
public onEvent(event: AnnotationLayerRenderedEvent): void {
  (event.source as any).div.querySelectorAll("input").forEach(x => x.disabled=true);
}
```

The type cast is necessary because I assigned the wrong type to the source attribute. The next version is going to fix this.

**Reply by wjmackinnon** [reply]
That fixed it, thank you... however, on this newest update, now, the default tool is the hand tool. Is there a way to change this back to the text selection tool? Thanks

**Reply by stephanrauh** [reply]
Wait, what? I didn't change that default, at least not knowingly. And I don't see how it might be caused by your custom `onEvent()` method. It's more likely you've klicked the hand tool button, so pdf.js stores it in the history and activates it every time you open the same PDF again. But I'm not sure - when I looked up the local storage of the browser, it doesn't contain the selection mode.

Be that as it may, you should be able to override the hand mode by setting the attribute `[handTool]="false"`.

<img width="676" alt="grafik" src="https://user-images.githubusercontent.com/3045767/216726134-e3e92287-9b77-44c8-9172-32dddc017f67.png">

Does this help you, or do you need more help from me?

Best regards,
Stephan

**Reply by stephanrauh** [reply]
I just saw there's also an option in the `pdfDefaultOptions`allowing you to disable forms. I didn't try it, but maybe it's the more elegant approach. `pdfDefaultOptions.annotationMode=0` should do the trick.

**Reply by wjmackinnon** [reply]
Hi there, I appreciate the response! Unfortunately, [handTool] does not seem to work in the latest version. Even when set to false, the default tool is still the hand tool. I will look into using that default options modifier to remove the annotations, however your other solution works great too.

**Reply by stephanrauh** [reply]
Oops. You're right, I've implemented a setter that doesn't set the property it claims to set. Version 16.0.1 fixes that.

**Comment by stephanrauh** [comment]
Your feature has landed: version 17.0.0-alpha.9 adds the flag `[disableForms]`.
